dds/idl/marshal_generator.cpp

```diff
@@ -1722,9 +2152,11 @@ namespace {
     type = resolveActualType(type);
     Classification fld_cls = classify(type);
 
-    const string qual = prefix + '.' + insert_cxx11_accessor_parens(name, is_union_member),
-          shift = prefix.substr(0, 2),
-          expr = qual.substr(3);
+    string qual = prefix + '.' + insert_cxx11_accessor_parens(name, is_union_member);
+    // if there is a stray '.' on the end, strip it off
+    if (qual[qual.length() - 1] == '.') qual.pop_back();
```

Comment:
```suggestion
    if (qual[qual.length() - 1] == '.') {
      qual.erase(qual.length() - 1);
    }
```

The C++03 build is complaining because 03 doesn't have `std::string::pop_back`.